### ISSUE TITLE
RCS1176 should be reported only for variables where type can be inferred

### ISSUE THREAD

**Issue by hankovich** [JosefPihrt/Roslynator#1133]
**Product and Version Used**: 4.3.0

**Actual Behavior**:
```csharp
bool TryParse<T>(string? s, out T t)
{
    t = default!;

    return false;
}

TryParse("wasted", out IntPtr i); // <- RCS1176: Use 'var' instead of explicit type, but IntPtr is inferred from this code
```
